src/Carousel.jsx

```diff
@@ -0,0 +1,100 @@
+/*
+heart icon in upper right corner to favorite- clicking fills in color
+below heart icon is expand image icon. containers for both are 36x36
+clicking on the image brings it out into a light box
+next/prev buttons are on far left & right side of image, vertically centered
+  circular white buttons with a drop shadow, arrows are black single chevrons (svg can be downloaded from etsy)
+  chevron box is 24x24px, button is 48x48
+  drop shadow slightly darkens on hover
+main image ~ 625px wide
+*/
+
+import React from 'react';
+import './styles/styles.css';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      favorite: false
+    };
+  }
+
+  favoriteItem() {
+    if (this.state.favorite) {
+      document.getElementById('fav-heart-icon').setAttribute('style', 'display: none;');
+      this.setState({favorite: false});
+    } else {
+      document.getElementById('fav-heart-icon').setAttribute('style', 'display: block;');
+      this.setState({favorite: true});
+    }
+  }
+
+  nextImg() {
+    let i = this.props.images.indexOf(this.props.currentImage);
+
+    if (this.currentImgIndex === -1) {
+      this.props.setCurrentImage(0);
+      console.log('An error occured getting the next image');
+    } else if (i === this.props.images.length - 1) {
+      this.props.setCurrentImage(0);
+    } else {
+      this.props.setCurrentImage(i + 1);
+    }
+  }
+
+  prevImg() {
+    let i = this.props.images.indexOf(this.props.currentImage);
+
+    if (this.currentImgIndex === -1) {
+      this.props.setCurrentImage(0);
+      console.log('An error occured getting the next image');
+    } else if (i === 0) {
+      this.props.setCurrentImage(this.props.images.length - 1);
+    } else {
+      this.props.setCurrentImage(i - 1);
+    }
+  }
```

Comment:
I like how safe these look by keeping the lone source of truth in the ancestor. 
Possibly the index changing logic could live up there if you wanted.